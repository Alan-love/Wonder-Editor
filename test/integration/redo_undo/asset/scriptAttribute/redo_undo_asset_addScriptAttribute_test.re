open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("redo_undo: asset add script attribute", () => {
    let sandbox = getSandboxDefaultVal();

    let _simulateTwiceAddScriptAttribute = () => {
      let assetTreeData =
        MainEditorAssetTreeTool.BuildAssetTree.buildEmptyAssetTree();

      MainEditorAssetHeaderOperateNodeTool.addScriptAttribute();
      MainEditorAssetHeaderOperateNodeTool.addScriptAttribute();
    };

    let _beforeEach = () => {
      MainEditorSceneTool.initState(~sandbox, ());
      MainEditorSceneTool.createDefaultScene(
        sandbox,
        MainEditorAssetTool.initAssetTree,
      );

      EventListenerTool.buildFakeDom()
      |> EventListenerTool.stubGetElementByIdReturnFakeDom;
    };

    let _afterEach = () => restoreSandbox(refJsObjToSandbox(sandbox^));

    beforeEach(() => sandbox := createSandbox());
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    RedoUndoTool.testRedoUndoTwoStep(
      sandbox,
      "prepare first step: set currentSceneTreeNode",
      (_simulateTwiceAddScriptAttribute, _beforeEach, _afterEach),
      BuildComponentTool.buildAssetChildrenNode,
    );
  });