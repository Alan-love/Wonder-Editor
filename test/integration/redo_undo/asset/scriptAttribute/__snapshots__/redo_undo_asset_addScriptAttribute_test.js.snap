// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: asset add script attribute prepare first step: set currentSceneTreeNode test redo operate test redo one step undo step which from second to zero, redo step which from zero to first 1`] = `
<article
  className="wonder-asset-assetChildren"
>
  <article
    className="wonder-asset-fileBox "
    onClick={[Function]}
  >
    <div
      className="box-image"
    >
      <img
        onDragStart={[Function]}
        src="./public/img/scriptAttribute.png"
      />
    </div>
    <div
      className="item-text "
    >
      <span>
        New Script Attribute
      </span>
    </div>
  </article>
</article>
`;

exports[`redo_undo: asset add script attribute prepare first step: set currentSceneTreeNode test redo operate test redo two step undo step which from second to zero,redo step which from zero to second 1`] = `
<article
  className="wonder-asset-assetChildren"
>
  <article
    className="wonder-asset-fileBox "
    onClick={[Function]}
  >
    <div
      className="box-image"
    >
      <img
        onDragStart={[Function]}
        src="./public/img/scriptAttribute.png"
      />
    </div>
    <div
      className="item-text "
    >
      <span>
        New Script Attribute
      </span>
    </div>
  </article>
  <article
    className="wonder-asset-fileBox "
    onClick={[Function]}
  >
    <div
      className="box-image"
    >
      <img
        onDragStart={[Function]}
        src="./public/img/scriptAttribute.png"
      />
    </div>
    <div
      className="item-text "
    >
      <span>
        New Script Attribute 1
      </span>
    </div>
  </article>
</article>
`;

exports[`redo_undo: asset add script attribute prepare first step: set currentSceneTreeNode test undo operate test not undo 1`] = `
<article
  className="wonder-asset-assetChildren"
>
  <article
    className="wonder-asset-fileBox "
    onClick={[Function]}
  >
    <div
      className="box-image"
    >
      <img
        onDragStart={[Function]}
        src="./public/img/scriptAttribute.png"
      />
    </div>
    <div
      className="item-text "
    >
      <span>
        New Script Attribute
      </span>
    </div>
  </article>
  <article
    className="wonder-asset-fileBox "
    onClick={[Function]}
  >
    <div
      className="box-image"
    >
      <img
        onDragStart={[Function]}
        src="./public/img/scriptAttribute.png"
      />
    </div>
    <div
      className="item-text "
    >
      <span>
        New Script Attribute 1
      </span>
    </div>
  </article>
</article>
`;

exports[`redo_undo: asset add script attribute prepare first step: set currentSceneTreeNode test undo operate test undo one step step which from second to first 1`] = `
<article
  className="wonder-asset-assetChildren"
>
  <article
    className="wonder-asset-fileBox "
    onClick={[Function]}
  >
    <div
      className="box-image"
    >
      <img
        onDragStart={[Function]}
        src="./public/img/scriptAttribute.png"
      />
    </div>
    <div
      className="item-text "
    >
      <span>
        New Script Attribute
      </span>
    </div>
  </article>
</article>
`;

exports[`redo_undo: asset add script attribute prepare first step: set currentSceneTreeNode test undo operate test undo two step step which from second to zero 1`] = `
<article
  className="wonder-asset-assetChildren"
/>
`;
